In [1]:
import pandas as pd

books = pd.read_csv("allbacks.csv")
books.head()

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb


In [2]:
import os
os.getcwd()

'/Users/anasty'

In [3]:
import glob
glob.glob("/Users/anasty/**/allbcks.csv", recursive=True)

[]

In [4]:
import os
[x for x in os.listdir("/Users/anasty") if x.endswith(".csv")]

['messy_sales.csv',
 'allbacks.csv',
 'usarrests.csv',
 'chickwts.csv',
 'sales_clean.csv',
 'bridges.csv']

In [5]:
print("Mean weight:", books["weight"].mean())
print("Mean weight by cover:")
print(books.groupby("cover")["weight"].mean())

print("Sorted covers:")
print(books["cover"].sort_values().tolist())

Mean weight: 706.6666666666666
Mean weight by cover:
cover
hb    796.428571
pb    628.125000
Name: weight, dtype: float64
Sorted covers:
['hb', 'hb', 'hb', 'hb', 'hb', 'hb', 'hb', 'pb', 'pb', 'pb', 'pb', 'pb', 'pb', 'pb', 'pb']


### Results

1. The median cover type is **pb (paperback)**.
   Paperback is the middle cover type.

2. The mean weight of all books is **706.67 grams**.
   Books weigh around 707 grams on average.

3. The mean weight for hardcover books is **796.43 grams**, while the mean weight for paperback books is **628.13 grams**.
   Hardcover books weigh more than paperback books on average.

### By-hand verification

I added all 15 book weights and divided the total by 15.

10600 / 15 = 706.67 grams

This matches the Python result.

In [6]:
books["weight"].sum(), len(books)

(np.int64(10600), 15)